docs/sidechain/spec.md

```diff
@@ -0,0 +1,945 @@
+chains themselves and collect the rewards themselves. Allowing the servers to be
+private had the advantage of greatly reducing the attack surface on these
+servers. They won't have to deal with adversarial input to their RPC commands,
+and since their ip address will be unknown, it will be hard to mount an DOS
+attack.
+
+In the second model, the witness server monitors events on a chain, but does not
+submit their signatures themselves. Instead, another party will pay the
+attestation server for their signature (for example, through a subscription
+fee), and the attestation server allows that party to collect the signer's
+reward. The account that the signer's reward goes to is part of the message that
+the witness server signs. In this model, it is likely that the witness server
+only listens to events on one chain. This model allows for a single witness
+server to act as a witness for multiple sidechains.
+
+As a side note, since submitting a signature requires submitting a transaction
+and paying a fee, supporting rewards for signatures is an important requirement.
+Of course, the reward can be higher than the fee, providing an incentive to
+running a witness server.
+
+## Why use the signer's list on the account
+
+The signatures that the witness servers use must match the signatures on that
+door's signer's list. But this isn't the only way to implement this. A sidechain
+ledger object could contain a signer's list that's independent from the door
+account. The reasons for using the door account signer's list are:
+
+1) The sidechain signer's list can be used to move funds from the account.
+   Putting this list on the door account emphasizes this trust model.
+   
+2) It allows for emergency action. If something goes very, very wrong, funds
+   could still be moved if the entities on the signer's list sign a regular
+   transaction.
+
+3) If the door account has multiple sidechains, a strange use-case, but one that
+   is currently supported. If the sidechain share a common asset type, the trust
+   model is the union of the signer's list. Keeping the signer's list on the
+   account makes this explicit.
+
+4) It's a more natural way to modify sidechain parameters.
+
+## Setting up a sidechain
+
+Setting up a sidechain requires the following:
+
+1) Create a new account for the door account (or use the root account - this can
+   be useful for sidechains). Note, that while more than one sidechain on a door account
+   is supported, is it discouraged.
+
+2) If necessary, setup the trust lines needed for IOUs.
+
+3) Create the sidechain ledger object on each door account.
+
+4) Enable multi-signatures on the two door accounts. These keys much match the
+   keys used by the witness servers. Note that the two door accounts may have
+   different multi-signature lists.
+
+5) Disable the master key, so only the keys on the multi-signature list can
+   control the account.
```

Comment:
Is this required by the witness servers, in the same way that it was with the original federators?

Reply:
No, it's not strictly required - it's not special like it was in the old design. But if you don't do this you're operating under a very different trust model.

Reply:
I was mainly asking with regard to debugging purposes - that was a tricky one to resolve.

Reply:
For debugging purposes you can leave the master key enabled, yes.